flake/default.nix

```diff
@@ -11,10 +11,5 @@
   perSystem = {pkgs, ...}: {
     formatter = pkgs.alejandra;
     pre-commit.settings.hooks.alejandra.enable = true;
-
-    # Neovim uses lua 5.1 as it is the version which supports JIT
-    _module.args = {
-      lua = pkgs.lua5_1;
```

Comment:
`lua = pkgs.luajit` is not enough ?

Reply:
I find it nice to have a high level lua you can override with alternative interpreters. There are many different lua interpreters like luau one might want to test. Also luajit might be available on fewer platforms than puc lua so if one can easily go back to puc lua, it's better.
Btw how would I override `_module.args` from my flake to run sthg like:"
```
_module.args = {
   lua = toto;
}
```
?

Reply:
Ok, I see the point for a user to be able to override the lua.
However, this snippet does not allow for such an external override.
Actually, it was not the case before.

What we could do to let the user override this is to add a `pkgs.luaNeovim` package that we use in the flake.
Maybe then, the user could override what `pkgs.luaNeovim` is on its side, but I am not entirely sure.

Reply:
Don't make it too complicated. Override the lua input in packages/neovim.nix, same as we do for tree-sitter, treesitter-parsers, etc.